### PR TITLE
RF02-TC-071: Verificar carga del listado de cargos

### PR DESCRIPTION
**Pull request by jdpapiero**
RF02-TC-071: Verificar carga del listado de cargos
1.- Ingresar al sistema con una cuenta valida de usuario válida
2.- Ingresar a la opción del menú "Estructura organizacional"
3.- Seleccionar la opción "Cargos del personal"
4.- Verificar que se haya cargado el listado con los cargos registrados en la base de datos.
5.- Verificar que el listado contenga paginador de contenido.

### CODE REVIEW

**Comment by jdpapiero** [comment]
image

**Comment by jdpapiero** [comment]
image

File: src/test/resources/features/Position/listPosition.feature [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@listPosition
+Feature: List Position
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar la carga del listado de los cargos con un paginador de la lista
+    Given Menu principal estee cargado
+    When Se selecciona la opcion del menu 'Estructura organizacional'
```
Comment (marcelogaray) [inline src/test/resources/features/Position/listPosition.feature]: When seleccionar 'Estructura Organizacional' en 'Menu Principal'

Comment (marcelogaray) [reply src/test/resources/features/Position/listPosition.feature]: Please review your scenario

File: src/test/resources/features/Position/listPosition.feature [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Position feature review the behavior in the feature
+# RF02-TC-071 UI > Asignación de funciones > Estructura organizacional >
+# Cargos de la empresa > Verificar la carga del listado de los cargos con un paginador de la lista
+
+@listPosition
```
Comment (marcelogaray) [inline src/test/resources/features/Position/listPosition.feature]: remove it please

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Position/Position.java [hunk 3]
```diff
@@ -9,7 +9,21 @@
...
+    private WebElement table;
+
+    @FindBy(xpath = "//div/div[contains(text(),'Items per page:')]")
+    private WebElement paginator;
+
     public Position() {
         CommonEvents.isVisible(title);
     }
+
+    public Boolean validListPositions() {
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Position/Position.java]: boolean

**Review by marcelogaray — CHANGES_REQUESTED** [review]
review observations

File: src/main/java/org/umssdiplo/automationv01/core/managepage/Position/Position.java [hunk 4]
```diff
@@ -9,7 +9,21 @@
     @FindBy(xpath = "//h1[contains(text(),'Cargos de la empresa')]")
     private WebElement title;
 
+    @FindBy(name = "tblPositions")
+    private WebElement table;
+
+    @FindBy(xpath = "//div/div[contains(text(),'Items per page:')]")
```
Comment (marcelogaray) [inline src/main/java/org/umssdiplo/automationv01/core/managepage/Position/Position.java]: it should be enough with //div[contains(text(),'Items per page:')]

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 5]
```diff
@@ -55,6 +59,26 @@ public void validarListaDeUsuarios() throws Throwable {
...
+
+    @Then("^Validar que exista el listado con los cargos registrados en la base de datos$")
+    public void validarQueExistaElListadoConLosCargosRegistradosEnLaBaseDeDatos() throws Throwable {
+        Assert.assertTrue(position.validListPositions(), "No se encuentra la lista de cargos");
+    }
+
+    @And("^Verificar que el listado contenga paginador de contenido$")
+    public void verificarQueElListadoContengaPaginadorDeContenido() throws Throwable {
+        Assert.assertTrue(position.validPaginator(), "No se encuentra el paginador");
+
```
Comment (marcelogaray) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: it should be careful with merge, in this point you need to add a close braces.

**Review by marcelogaray — COMMENTED** [review]
review my observations

**Comment by marcelogaray** [comment]
we are going to close it, please let me know when you will fix your conflicts(merge with development)

**Comment by jdpapiero** [comment]
Commit with merge and improves recomended

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 6]
```diff
@@ -59,14 +64,44 @@ public void validarListaDeUsuarios() throws Throwable {
...
+    }
+
+    @And("^seleccionar la opcion 'Cargos del personal' en el sub menu de estructura organizacional$")
+    public void seleccionarLaOpcionCargosDelPersonalEnElSubMenuDeEstructuraOrganizacional() throws Throwable {
+        position = subMenuOrganizationalStructure.clickSubMenuPositions();
+    }
+
+    @Then("^validar que exista el listado con los cargos registrados en la base de datos$")
+    public void validarQueExistaElListadoConLosCargosRegistradosEnLaBaseDeDatos() throws Throwable {
+        Assert.assertTrue(
```
Comment (marcelogaray) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: just one line please

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 7]
```diff
@@ -59,14 +64,44 @@ public void validarListaDeUsuarios() throws Throwable {
...
+                position.validListPositions(),
+                String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Positions list")
+        );
+    }
+
+    @And("^verificar que el listado contenga paginador de contenido$")
+    public void verificarQueElListadoContengaPaginadorDeContenido() throws Throwable {
+        Assert.assertTrue(
```
Comment (marcelogaray) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: just on line

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java [hunk 8]
```diff
@@ -59,14 +64,44 @@ public void validarListaDeUsuarios() throws Throwable {
...
+                position.validPaginator(),
+                String.format(ErrorMessage.ERROR_MESSAGE_ELEMENT_VISIBLE, "Positions paginator list")
+        );
+        Assert.assertTrue(position.validPaginator(), "No se encuentra el paginador");
+
```
Comment (marcelogaray) [inline src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: remove blank line

File: src/test/resources/features/Position/listPosition.feature [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar la carga del listado de los cargos del personal
+    Given 'Menu principal' estee cargado
+    When seleccionar la opcion 'Estructura organizacional' en el menu principal
+    And seleccionar la opcion 'Cargos del personal' en el sub menu de estructura organizacional
+    Then validar que exista el listado con los cargos registrados en la base de datos
```
Comment (marcelogaray) [inline src/test/resources/features/Position/listPosition.feature]: "en la base de datos"? is it correct? this means that you will connect to DB and you will do a validation of the information that you will have, and finally compare with UI, is it correct?

File: src/test/resources/features/Position/listPosition.feature [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    And verificar que el listado contenga paginador de contenido
```
Comment (marcelogaray) [inline src/test/resources/features/Position/listPosition.feature]: please use 'verificar' or 'validar' just to have an standard, in this case I suggest to use: And verificar que 'Paginador' se muestra debajo de la tabla 'Lista de Cargo'

**Review by marcelogaray — CHANGES_REQUESTED** [review]
review observations.

Please once you finish it send me a pull request, in the meantime I will close it.

File: src/test/resources/features/Position/listPosition.feature (hunk 10 above)
Comment (jdpapiero) [reply src/test/resources/features/Position/listPosition.feature]: I take your recomendation, thanks

File: src/test/resources/features/Position/listPosition.feature (hunk 9 above)
Comment (jdpapiero) [reply src/test/resources/features/Position/listPosition.feature]: I rewrite the step line

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java (hunk 8 above)
Comment (jdpapiero) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: done

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java (hunk 7 above)
Comment (jdpapiero) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: done

File: src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java (hunk 6 above)
Comment (jdpapiero) [reply src/test/java/org/umssdiplo/automationv01/stepdefinitionproject/StepsDefinitionSSID.java]: done

**Comment by jdpapiero** [comment]
Reopened with fix the changes recomended

File: src/test/resources/features/Position/listPosition.feature [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+  Background: Pagina de inicio es mostrada una vez que se ponen los credenciales correctos de inicio de sesión
+    Given 'SSI-D' pagina de inicio de sesion es cargada
+    And Ingresar los credenciales validos en la pagina 'Login' para ingresar al sistema
+
+  Scenario: Verificar la carga del listado de los cargos del personal
+    Given 'Menu principal' estee cargado
+    When seleccionar la opcion 'Estructura organizacional' en el menu principal
+    And seleccionar la opcion 'Cargos del personal' en el sub menu de estructura organizacional
+    Then validar que se cargue el listado de cargos en la pagina 'Cargos de la empresa'
```
Comment (marcelogaray) [inline src/test/resources/features/Position/listPosition.feature:15]: you should use validar or verificar

**Review by marcelogaray — CHANGES_REQUESTED** [review]
review observation
Please once you finish it send me a pull request, in the meantime I will close it.

File: src/test/resources/features/Position/listPosition.feature (hunk 11 above)
Comment (jdpapiero) [reply src/test/resources/features/Position/listPosition.feature:15]: change for just use 'validar'

**Comment by jdpapiero** [comment]
change for just use 'validar'

**Review by marcelogaray — APPROVED** [review]

**Pull request merged**